### PR TITLE
Use the new `pipe` package for running external commands

### PR DESCRIPTION
**Pull request by mhagger**
The existing code wasn't always checking errors and freeing resources correctly. It's quite difficult, especially to handle all error paths correctly.

So instead of "handcrafting" command pipelines using low-level facilities like `exec.Cmd`, switch to using the `pipe` package that was introduced in #90. That package takes care of most of the details internally, presenting a more convenient and less error-prone UI.

This change clears up a bunch of linter errors regarding unchecked errors.

Note that this PR branch is built on top of the changes that were added in PRs #89 and #90. Those PRs can be merged separately or in combination with this one.

/cc @tgummerer and @ttaylorr as possible reviewers

### CODE REVIEW

File: git/obj_iter.go [hunk 1]
```diff
@@ -2,128 +2,147 @@ package git
...
-		return nil, nil, err
-	}
-
-	out2, err := cmd2.StdoutPipe()
-	if err != nil {
-		in2.Close()
-		out1.Close()
-		cmd1.Wait()
-		return nil, nil, err
+func (repo *Repository) NewObjectIter(ctx context.Context, args ...string) (*ObjectIter, error) {
```
Comment (tgummerer) [inline git/obj_iter.go:25]: The `args` object seems to be unused now, with `--stdin` and `--date-order` hardcoded in the rev list command.  I don't really see the need to pass `args` in here, should that just be dropped from the signature?

**Review by tgummerer — DISMISSED** [review]
One small question about a function argument that seems unused inside, but other than that looks good to me.  Thanks as always for splitting this up into digestible commits.

File: git/obj_iter.go (hunk 1 above)
Comment (mhagger) [reply git/obj_iter.go]: Nice catch! Fixed.

**Review by tgummerer — APPROVED** [review]
Thanks!

**Pull request merged**